st2actions/st2actions/runners/actionchainrunner.py

```diff
@@ -341,26 +333,43 @@ def _resolve_params(action_node, original_parameters, results, chain_vars):
         context.update(chain_vars)
         context.update({RESULTS_KEY: results})
         context.update({SYSTEM_KV_PREFIX: KeyValueLookup()})
-        rendered_params = jinja_utils.render_values(mapping=action_node.params, context=context)
+        context.update({ACTION_KV_PREFIX: chain_context})
+        try:
+            rendered_params = jinja_utils.render_values(mapping=action_node.params,
+                                                        context=context)
+        except Exception as e:
+            LOG.exception('Jinja rendering failed.')
+            raise ParameterRenderingFailedException(e)
         LOG.debug('Rendered params: %s: Type: %s', rendered_params, type(rendered_params))
         return rendered_params
 
-    def _run_action(self, action_node, parent_context, params, wait_for_completion=True):
-        liveaction = LiveActionDB(action=action_node.ref)
-        liveaction.parameters = action_param_utils.cast_params(action_ref=action_node.ref,
-                                                               params=params)
+    def _run_next_action(self, action_node, parent_context, action_params, context_result,
```

Comment:
Why _run_next_action as opposed to _run_action or _run_task? The next part seems to have been handled outside.